sanic/cookies.py

```diff
@@ -104,13 +103,19 @@ def __setitem__(self, key, value):
     def encode(self, encoding):
         output = ['%s=%s' % (self.key, _quote(self.value))]
         for key, value in self.items():
-            if key == 'max-age' and isinstance(value, int):
-                output.append('%s=%d' % (self._keys[key], value))
-            elif key == 'expires' and isinstance(value, datetime):
-                output.append('%s=%s' % (
-                    self._keys[key],
-                    value.strftime("%a, %d-%b-%Y %T GMT")
-                ))
+            if key == 'max-age':
+                try:
```

Comment:
>Also the previous code and this one is not semantically the same. In previous case, if key=='max-age' and value in a str, the if block will not be executed.

Yes, but I'm saying it should be. At that point, you know the cookie key is expires or max-age -- why continue to compare it to other things, when you can just format it one way and then fallback if it fails? This results in fewer comparisons in the presence of those keys, and hence approximately %4 speedup.